lib/kubernetes/events/kubernetes_event_monitor.rb

```diff
@@ -0,0 +1,40 @@
+class KubernetesEventMonitor
+  def initialize(api_endpoint, api_version)
+    @api_endpoint = api_endpoint
+    @api_version = api_version
+  end
+
+  def inventory
+    require 'kubeclient'
+    @inventory ||= Kubeclient::Client.new(@api_endpoint, @api_version)
+  end
+
+  def watcher(version = nil)
+    @watcher ||= inventory.watch_events(version)
+  end
+
+  def start
+    trap(:TERM) { $kube_log.info('EventMonitor#start: ignoring SIGTERM') }
+    @inventory = nil
+    @watcher = nil
+  end
+
+  def stop
+    watcher.finish
```

Comment:
Hrm, I'm not sure of the question...

If we change the endpoint on the EMS object, then I'm not sure we do anything to the workers.  And, we probably should.  Sounds like an additional PR unrelated to this one, though.

I guess I can't see another way for the endpoint to change.